__tests__/topics.test.js

```diff
@@ -232,32 +231,35 @@ describe("/api/articles/:article_id/comments", () => {
       .then(({ body }) => {
         expect(body.comment).toMatchObject({
           comment_id: expect.any(Number),
-          body: expect.any(String),
-          article_id: expect.any(Number),
-          author: expect.any(String),
+          body: "cool comment",
+          article_id: 9,
+          author: "icellusedkars",
           votes: expect.any(Number),
           created_at: expect.any(String),
         });
       });
   });
-  test("POST: 201, responds with the requested comment", () => {
+  test("GET 400: responds with 400 if request is NaN", () => {
     const comment = { username: "icellusedkars", body: "cool comment" };
     return supertest(app)
-      .post("/api/articles/9/comments")
+      .post("/api/articles/NaN/comments")
       .send(comment)
-      .expect(201)
+      .expect(400)
       .then(({ body }) => {
-        expect(body.comment).toMatchObject({
-          comment_id: 19,
-          body: "cool comment",
-          article_id: 9,
-          author: "icellusedkars",
-          votes: 0,
-          created_at: expect.any(String),
-        });
+        expect(body).toEqual({ msg: "Bad Request" });
       });
   });
-  test("GET 400: responds with 400 if request is NaN", () => {
+  test("GET 404: responds with 404 if article id doesn't exist", () => {
+    const comment = { username: "icellusedkars", body: "cool comment" };
+    return supertest(app)
+      .post("/api/articles/12345/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body).toEqual({ msg: "Bad Request" });
+      });
+  });
+  test("GET 404: responds with 404 if username doesn't exist", () => {
```

Comment:
The test doesnt match the description - the username you've is valid - it still has NaN in the path and the expected code seems to be 400, not 404